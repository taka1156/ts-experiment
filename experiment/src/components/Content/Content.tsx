import React from 'react';

type ContentProps = React.ComponentPropsWithRef<'textarea'>;

const Content = (props?: ContentProps) => {
  return (
    <textarea
      {...props}
      defaultValue={props?.defaultValue || '何も設定してません'}
      data-testid='inputTarget'
    ></textarea>
  );
};

export { Content };
