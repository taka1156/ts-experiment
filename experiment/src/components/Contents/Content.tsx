import React from 'react';

type ContentProps = React.ComponentPropsWithRef<'textarea'>;

const Content = (props: ContentProps) => {
  props.value = '初期の入力値（JS経由）';
  return <textarea {...props} data-testid='inputTarget'></textarea>;
};

export { Content };
