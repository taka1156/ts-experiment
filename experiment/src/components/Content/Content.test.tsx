import { ReactElement } from 'react';
import { describe, expect, it, afterEach } from 'vitest';
import userEvent from '@testing-library/user-event';
import { cleanup, render, screen } from '@testing-library/react';
import { Content } from './Content';

const setup = (jsx: ReactElement) => {
  return {
    user: userEvent.setup(),
    dom: render(jsx)
  };
};

describe('textareaテスト', () => {
  afterEach(() => {
    cleanup();
  });

  it('スナップショット', () => {
    const args: React.ComponentPropsWithRef<'textarea'> = {
      defaultValue: 'test'
    };
    const { dom } = setup(<Content {...args} />);

    expect(dom).toMatchSnapshot();
  });

  it('input', async () => {
    const { user, dom } = setup(<Content />);

    const input = screen.getByTestId<HTMLTextAreaElement>('inputTarget');

    const testInputValue = 'テスト入力';
    await user.clear(input);
    await user.type(input, testInputValue);

    expect(input.value).toBe(testInputValue);

    expect(dom).toMatchSnapshot();
  });

  it('バリデーション', async () => {
    const { user, dom } = setup(<Content />);

    const input = screen.getByTestId<HTMLTextAreaElement>('inputTarget');

    await user.clear(input);
    await user.tab(); // = onBlur

    const error = screen.getByTestId<HTMLParagraphElement>('errorTarget');

    expect(error.textContent).toBe('必須入力です');

    expect(dom).toMatchSnapshot();
  });
});
