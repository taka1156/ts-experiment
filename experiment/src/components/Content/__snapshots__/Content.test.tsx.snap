// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`textareaテスト > input 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <textarea
        data-testid="inputTarget"
      />
    </div>
  </body>,
  "container": <div>
    <textarea
      data-testid="inputTarget"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`textareaテスト > スナップショット 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <textarea
        data-testid="inputTarget"
      >
        test
      </textarea>
    </div>
  </body>,
  "container": <div>
    <textarea
      data-testid="inputTarget"
    >
      test
    </textarea>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`textareaテスト > バリデーション 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <textarea
        data-testid="inputTarget"
      />
      <p
        data-testid="errorTarget"
      >
        必須入力です
      </p>
    </div>
  </body>,
  "container": <div>
    <textarea
      data-testid="inputTarget"
    />
    <p
      data-testid="errorTarget"
    >
      必須入力です
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
